File: CS/WpfApplication2/ViewModel.cs
using System;
using System.Collections.ObjectModel;
using DevExpress.Mvvm;

namespace WpfApplication2 {
	public class ViewModel : ViewModelBase {
		public ObservableCollection<TestData> List {
			get { return GetValue<ObservableCollection<TestData>>(); }
			set { SetValue(value); }
		}

		public ViewModel() {
			List = new ObservableCollection<TestData>();
			GenerateData(20);
		}

		private void GenerateData(int objectCount) {
			for (int i = 0; i < objectCount; i++)
				List.Add(new TestData() { Id = Guid.NewGuid(), Number = i });
		}
	}
	public class TestData {
		public Guid Id { get; set; }
		public int Number { get; set; }
		public bool IsChecked { get; set; }
	}
}

File: VB/WpfApplication2/ViewModel.vb
Imports System
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm

Namespace WpfApplication2
	Public Class ViewModel
		Inherits ViewModelBase

		Public Property List() As ObservableCollection(Of TestData)
			Get
				Return GetValue(Of ObservableCollection(Of TestData))()
			End Get
			Set(ByVal value As ObservableCollection(Of TestData))
				SetValue(value)
			End Set
		End Property

		Public Sub New()
			List = New ObservableCollection(Of TestData)()
			GenerateData(20)
		End Sub

		Private Sub GenerateData(ByVal objectCount As Integer)
			For i As Integer = 0 To objectCount - 1
				List.Add(New TestData() With {
					.Id = Guid.NewGuid(),
					.Number = i
				})
			Next i
		End Sub
	End Class
	Public Class TestData
		Public Property Id() As Guid
		Public Property Number() As Integer
		Public Property IsChecked() As Boolean
	End Class
End Namespace